client/extensions/woocommerce/app/promotions/promotion-update.js

```diff
@@ -6,36 +6,211 @@
 
 import React from 'react';
 import PropTypes from 'prop-types';
+import { bindActionCreators } from 'redux';
 import { connect } from 'react-redux';
 import { localize } from 'i18n-calypso';
+import page from 'page';
+import { debounce } from 'lodash';
 
 /**
  * Internal dependencies
  */
 import Main from 'components/main';
+import accept from 'lib/accept';
+import { successNotice, errorNotice } from 'state/notices/actions';
+import { getLink } from 'woocommerce/lib/nav-utils';
+import {
+	fetchPromotions,
+	updatePromotion,
+	deletePromotion,
+} from 'woocommerce/state/sites/promotions/actions';
+import { fetchProductCategories } from 'woocommerce/state/sites/product-categories/actions';
+import { getProductCategories } from 'woocommerce/state/sites/product-categories/selectors';
+import { editPromotion, clearPromotionEdits } from 'woocommerce/state/ui/promotions/actions';
 import { getSelectedSiteWithFallback } from 'woocommerce/state/sites/selectors';
+import { fetchSettingsGeneral } from 'woocommerce/state/sites/settings/general/actions';
+import { getPaymentCurrencySettings } from 'woocommerce/state/sites/settings/general/selectors';
+import {
+	getPromotionWithLocalEdits,
+	getPromotionableProducts,
+} from 'woocommerce/state/selectors/promotions';
+import { isValidPromotion } from './helpers';
+import PromotionHeader from './promotion-header';
+import PromotionForm from './promotion-form';
 
 class PromotionUpdate extends React.Component {
 	static propTypes = {
+		className: PropTypes.string,
```

Comment:
Could also add in `currency`, `products` and `productCategories`

Reply:
Added.